### PR TITLE
DBT-724 DBT-725 Workspace PR

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

This PR contains mainly the create workspace, edit and delete workspace you can also visualize the workspace list

## Test

- Login as an admin
- Go to create a workspace by clicking in the sidebar 
- Try to create a workspace edit it and delete it

### CODE REVIEW

File: src/modules/resources/components/resources/add-workspace-modal.vue [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      default: ''
+    },
+    limit: {
+      type: Number,
+      default: 5
+    },
+    loading: {
+      type: Boolean,
+      default: false
+    },
```
Comment (abelbordonado) [inline src/modules/resources/components/resources/add-workspace-modal.vue:68]: I dont think we need loading as a prop. Is shall be internal to the modal no? Loading adding user.

File: src/modules/resources/components/resources/add-workspace-modal.vue [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  components: { NameFieldInput },
+  props: {
+    title: {
+      type: String,
+      default: ''
+    },
+    limit: {
+      type: Number,
+      default: 5
+    },
```
Comment (abelbordonado) [inline src/modules/resources/components/resources/add-workspace-modal.vue:64]: ? Not required here right?

File: src/modules/resources/components/resources/add-workspace-modal.vue [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    loading: {
+      type: Boolean,
+      default: false
+    },
+    item: {
+      type: Object,
+      default: null
+    },
```
Comment (abelbordonado) [inline src/modules/resources/components/resources/add-workspace-modal.vue:72]: Is the item a workspace? Lets call it workspace then

File: src/modules/resources/components/resources/name-field-input.vue [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      :filled="true"
+      required
+      clearable
+      @input="$emit('input', $event)"
+    >
+    </v-text-field>
+  </ValidationProvider>
+</template>
+
+<script>
```
Comment (abelbordonado) [inline src/modules/resources/components/resources/name-field-input.vue:23]: I think we can use the FieldInput common component (as this component has nothing special itself.


image

File: src/modules/workspaces/workspace-list/components/NameFieldInput.vue [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      :error-messages="errors"
+      label="Workspace Name"
+      :filled="true"
+      required
+      clearable
+      @input="$emit('input', $event)"
+    >
+    </v-text-field>
+  </ValidationProvider>
+</template>
```
Comment (abelbordonado) [inline src/modules/workspaces/workspace-list/components/NameFieldInput.vue:21]: Duplicated code, I think you added it in common components.

Anyway use the FieldInput general one for the name

File: src/modules/workspaces/workspace-list/components/workspace-list.vue [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+      <!-- ------------ SLOTS ------------ -->
+      <template v-slot:[`item.name`]="{ item }">
+        <div class="d-flex align-center">
+          <span
+            :style="{ backgroundColor: item.color }"
+            class="circle mr-1"
+          ></span>
+          {{ item.name }}
+        </div>
+      </template>
```
Comment (abelbordonado) [inline src/modules/workspaces/workspace-list/components/workspace-list.vue:59]: We dont need to render the circle! this is from groups, can remove this template

File: src/modules/workspaces/workspace-list/components/workspace-list.vue [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+      if (!item) {
+        return
+      }
+      const result = await this.$swal.fire({
+        toast: true,
+        width: '400px',
+        icon: 'question',
+        title: 'ELIMINAR Workspace',
+        html: '<b>Esta acción es irreversible</b><br>¿Seguro que deseas eliminar este Workspace? Todos los alumnos seran dados de baja y perderas el histórico del Workspace',
+        showConfirmButton: true,
```
Comment (abelbordonado) [inline src/modules/workspaces/workspace-list/components/workspace-list.vue:192]: Esta acción es irreversible</b><br>¿Seguro que deseas eliminar este Workspace? Todos los materiales seran borrados del servidor y los alumnos no podrán acceder a ellos',

File: src/services/WorkspaceRepository.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+    if (response.status === 409) {
+      ResourceService.warning({
+        response,
+        title: 'Información Duplicada',
+        message: 'Ya existe un grupo con el mismo código o color'
+      })
+
+      return false
+    }
```
Comment (abelbordonado) [inline src/services/WorkspaceRepository.js:22]: No need, workspace can be duplicated.

**Review by abelbordonado — CHANGES_REQUESTED** [review]
Great Job!
Requested some changes! Remember to remove unwanted things on copy-paste.

File: src/modules/resources/components/resources/name-field-input.vue (hunk 4 above)
Comment (kayiganwa04) [reply src/modules/resources/components/resources/name-field-input.vue:23]: This field is not working well because it has some email restriction

**Comment by abelbordonado** [comment]
image


Create Workspace is calling to EDIT workspace now. So I can not create a new workspace

File: src/modules/resources/components/resources/add-workspace-modal.vue [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+              </v-btn>
+              <v-btn
+                dark
+                color="blue darken-1"
+                class="button flex-grow-1"
+                large
+                :loading="loading"
+                @click="onCreateWorkspace"
+              >
+                {{ item ? 'Edit' : 'Create' }}
```
Comment (abelbordonado) [inline src/modules/resources/components/resources/add-workspace-modal.vue:40]: Editar
Crear

File: src/modules/resources/components/resources/add-workspace-modal.vue [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+    <v-dialog
+      v-model="isOpen"
+      max-width="400px"
+      @close="onClose"
+    >
+      <validation-observer ref="formCreateWorkspace">
+        <v-card class="d-flex flex-column" >
+          <v-container class="pa-3">
+            <v-card-title class="d-flex justify-space-between pt-0 px-0">
+              <span class="text-h6 font-weight-bold">Create Workspace</span>
```
Comment (abelbordonado) [inline src/modules/resources/components/resources/add-workspace-modal.vue:12]: Crear

File: src/modules/workspaces/workspace-list/components/workspace-list.vue [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+          :can-rendering-header="$vuetify.breakpoint.width < 700"
+        />
+
+        <!-- ------------ ACTIONS ------------ -->
+        <v-toolbar flat class="indigo lighten-5 my-2" outlined>
+          <resource-title-toolbar-datatable title-text="Workspace" />
+
+          <v-spacer />
+
+          <ResourceButtonAdd text-button="Create Workspace" @click="onCreate" />
```
Comment (abelbordonado) [inline src/modules/workspaces/workspace-list/components/workspace-list.vue:29]: Crear

**Comment by abelbordonado** [comment]
image

**Review by abelbordonado — APPROVED** [review]
Amazing! now looks great 👍 

Lets do it 🚀

**Pull request merged**